### PR TITLE
MinifyImportsAndExports: Minify the memory and table as well.

### PR DESCRIPTION
**Pull request by kripken**
We were careful not to minify those, as well as the stack pointer, which
makes sense in dynamic linking. But we don't run this pass in dynamic linking
anyhow - we need the proper names of symbols in that case. So this was
not helping us, and was just a leftover from an early state.

This both a useful optimization and also important for https://github.com/WebAssembly/binaryen/issues/3043,
as the wasm backend exports the table as `__indirect_function_table` - a much
longer name than emscripten's `table`. So just changing to that would regress
code size on small projects. Once we land this, the name won't matter as it will
be minified anyhow.

This will break some tests on the roller as it improves code size beyond
current expectations, so it will require a double-roll or a temporary test
disabling.

### CODE REVIEW

File: src/passes/MinifyImportsAndExports.cpp [hunk 1]
```diff
@@ -172,6 +168,12 @@ struct MinifyImportsAndExports : public Pass {
         process(curr->base);
       }
     };
+    if (module->memory.exists) {
+      processImport(&module->memory);
+    }
+    if (module->table.exists) {
+      processImport(&module->table);
+    }
```
Comment (sbc100) [inline src/passes/MinifyImportsAndExports.cpp:176]: Could we just use `iterImportedMemories` and `iterImportedTables`?

Or better still add an `iterImports` and just use that?

Comment (kripken) [reply src/passes/MinifyImportsAndExports.cpp:176]: Ah, good idea for `iterImports`, that's much nicer. Added now.

File: src/passes/MinifyImportsAndExports.cpp [hunk 2]
```diff
@@ -148,10 +148,6 @@ struct MinifyImportsAndExports : public Pass {
     std::map<Name, Name> oldToNew;
     std::map<Name, Name> newToOld;
     auto process = [&](Name& name) {
-      // do not minifiy special imports, they must always exist
-      if (name == MEMORY_BASE || name == TABLE_BASE || name == STACK_POINTER) {
```
Comment (MaxGraey) [inline src/passes/MinifyImportsAndExports.cpp:152]: seems it minify `STACK_POINTER` as well. Is it supposed to be?

Comment (kripken) [reply src/passes/MinifyImportsAndExports.cpp:152]: Yes, the special stack pointer import is relevant only for dynamic linking, but as mentioned earlier, we don't run this pass in that case anyhow.

Comment (MaxGraey) [reply src/passes/MinifyImportsAndExports.cpp:152]: Yes, I see

**Pull request merged**